Improve `inspect` stubs

AlexWaygood opened the pull request.

JelleZijlstra commented:
I'm not sure making the Signature class generic is really helpful, and the mypy-primer output agrees. It's usually used for introspection, and most code doing that probably doesn't benefit from precise types for the return annotation.

It's also not always correct: your code will infer something like `Signature[int]` for `inspect.signature(lambda: 1)`, but in fact the signature that computes has no return_annotation.

AlexWaygood commented:
> It's also not always correct: your code will infer something like `Signature[int]` for `inspect.signature(lambda: 1)`, but in fact the signature that computes has no return_annotation.

Good point -- I've reverted the changes making it generic. I've also removed the TODO comment, since, from what you're saying (and I agree), it sounds like we _can't_ really be more specific here.

AlexWaygood commented on stdlib/inspect.pyi:
```diff
@@ -166,7 +167,6 @@ class Signature:
     empty = _empty
     @property
     def parameters(self) -> types.MappingProxyType[str, Parameter]: ...
-    # TODO: can we be more specific here?
```
Yeah
